docs/integrations/Instructions.md

```diff
@@ -0,0 +1,160 @@
+funds to.
+  - `data`        (Vec<u8>):                                  The data to call
+the target with.
+
+If the network is Serai, this is a transfer. Else, it's a withdrawal to the
+specified address with the specified data. Asset contracts perform no validation
+on these fields.
+
+### Shorthand
+
+All In Instructions are encoded as Shorthand. Shorthand is an enum which expands
+to an In Instruction.
+
+##### Raw
+
+Raw Shorthand encodes a raw In Instruction with no further processing.
+
+##### Swap
+
+  - `origin`  (Option<Address>): In Instruction's `origin`.
+  - `coin`    (Coin):            Coin to swap funds for.
+  - `minimum` (Amount):          Minimum amount of `coin` to receive.
+  - `out`     (Out Instruction): Final destination for funds.
+
+which expands to:
+
+```
+In Instruction {
+  origin,
+  target: Router,
+  data:   swap(Incoming Asset, out, minimum)
+}
+```
+
+where `swap` is a function which:
+
+  1) Swaps the incoming funds for SRI.
+  2) Swaps the SRI for `coin`.
+  3) Checks the amount of `coin` received is greater than `minimum`.
+  4) Executes `out` with the amount of `coin` received.
+
+For a Bitcoin to Monero swap, Swap Shorthand is expected to generally take:
+
+  - 1 byte to identify as Swap.
+  - 1 byte to not override `origin`.
+  - 1 byte for `coin`.
+  - 4 bytes for `minimum`.
+  - 1 byte for `out`'s `destination` field label.
+  - 1 byte for `out`'s `destination`'s ordinal byte.
+  - 65 bytes for `out`'s `destination`'s address.
+
+Or 74 bytes.
+
+##### Add Liquidity
+
+  - `origin`  (Option<Address>): In Instruction's `origin`.
+  - `minimum` (Amount):  Minimum amount of SRI to receive.
+  - `gas`     (Amount):  Amount of SRI to send to `address` to cover gas in the
+future.
+  - `address` (Address): Account to give the created liquidity tokens.
+
+which expands to:
+
+```
+In Instruction {
+  origin,
+  target: Router,
+  data:   swap_and_add_liquidity(Incoming Asset, address, minimum, gas)
+}
+```
+
+where `swap_and_add_liquidity` is a function which:
+
+  1) Swaps half of the incoming funds for SRI.
+  2) Checks the amount of SRI received is greater than `minimum`.
+  3) Calls `swap_and_add_liquidity` with the amount of SRI received - `gas`, and
+a matching amount of the incoming asset.
+  4) Transfers any leftover funds to `address`.
+
+For adding liquidity from Bitcoin, Add Liquidity Shorthand is expected to
+generally take:
+
+  - 1 byte to identify as Add Liquidity.
+  - 1 byte to not override `origin`.
+  - 5 bytes for `minimum`.
+  - 1/4 bytes for `gas`.
+  - 32 bytes for `address`.
+
+Or 40/43 bytes, depending on whether or not the Serai address already has gas.
+
+### Examples
+
+All examples are assumed to be from Bitcoin.
+
+##### Pong Example
+
+```
+In Instruction {
+  target: Bitcoin Asset Contract,
+  data:   Withdraw(Out Instruction { destination: Native(Bitcoin Sender) })
+}
+```
+
+would cause the created seraiBTC to be transferred to the Bitcoin Asset Contract
```

Comment:
sBTC is synth BTC, which is not what we have. sriBTC isn't optimal. I do prefer the long form seraiBTC, with a token symbol of simply BTC on Serai.